src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -506,7 +512,13 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
   model::timeout_clock::duration timeout) {
     auto maybe_tx = co_await stm->get_actual_tx(tx_id);
 
-    if (!maybe_tx) {
+    if (!maybe_tx.has_value()) {
+        if (maybe_tx.error() != tm_stm::op_status::not_found) {
+            co_return init_tm_tx_reply{.ec = tx_errc::unknown_server_error};
+        }
+    }
+
+    if (!maybe_tx.has_value()) {
```

Comment:
just want to confirm this isn't a mistake since there are two identical conditions one right after the other...

Reply:
this isn't a mistake but I agree it looks suspicious, I'll collapse them together